### PR TITLE
Documentation and cleanup

### PR DESCRIPTION
**Pull request by JaceCear**

### CODE REVIEW

File: src/animation_commands.c [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #include "global.h"
 #include "lib/m4a.h"
 
+#include "packed.h"
```
Comment (freshollie) [inline src/animation_commands.c:4]: Where is this used?

File: src/bg_triangles.c [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#include <stdlib.h> // abs()
```
Comment (freshollie) [inline src/bg_triangles.c:1]: Is this really needed for abs? Cos the `abs` optimisation is automatically handled by agbcc, I guess just for the types?

File: src/game/math_1.c [hunk 3]
```diff
@@ -165,7 +174,14 @@ void sub_8085CC0(UNK_8085D14 *p1)
     UNK_8085D14 *curr = p1;
     // memset
     for (i = 0; i < 32; i += 4) {
+#ifndef NON_MATCHING
         *((u32 *)curr)++ = 0;
+#else
+        u32 *p = (u32 *)curr;
```
Comment (freshollie) [inline src/game/math_1.c:180]: Wy is this more accurate?

File: src/game/multiplayer/mp_player.c [hunk 4]
```diff
@@ -750,7 +750,7 @@ void sub_801707C(void)
                             == SA2_ANIM_CHAR(SA2_CHAR_ANIM_21, CHARACTER_TAILS)
                         || s->graphics.anim
                             == SA2_ANIM_CHAR(SA2_CHAR_ANIM_20, CHARACTER_TAILS))
-                    && ((!!GRAVITY_IS_INVERTED != !(mpp->unk54 & 8))))) {
+                    && ((GRAVITY_IS_INVERTED ? 1 : 0) != !(mpp->unk54 & 8)))) {
```
Comment (freshollie) [inline src/game/multiplayer/mp_player.c:753]: Not a fan of this change

File: src/game/stage/background/zone_3.c [hunk 5]
```diff
@@ -138,7 +138,7 @@ void StageBgUpdate_Zone3Acts12(s32 a, s32 b)
     Vec2_16 *cursorStack;
     u8 *cursor;
     s32 pFlags;
-#ifdef NON_MATCHING
```
Comment (freshollie) [inline src/game/stage/background/zone_3.c:141]: How was this compiling before?

File: src/game/stage/background/zone_7_2.c [hunk 6]
```diff
@@ -20,7 +20,7 @@ NONMATCH("asm/non_matching/game/stage/background/StageBgUpdate_Zone7Acts12.inc",
 {
     // NOTE: GCC-Hack
     // u32 act = (gCurrentLevel - LEVEL_INDEX(ZONE_7, ACT_1)) % 2
-    u32 act = !!(gCurrentLevel ^ (LEVEL_INDEX(ZONE_7, ACT_1)));
+    u32 act = (gCurrentLevel ^ (LEVEL_INDEX(ZONE_7, ACT_1))) ? ACT_2 : ACT_1;
```
Comment (freshollie) [inline src/game/stage/background/zone_7_2.c:23]: Nice :D

File: src/game/stage/game_7.c [hunk 7]
```diff
@@ -84,11 +84,18 @@ void sub_802DBC0(u8 p0, u16 p1)
...
     register u32 r7 asm("r7") = p0;
     u8 *bgOffsets = gBgOffsetsHBlank;
-    s32 r2;
     register s32 r5 asm("r5") = 0;
     register u16 r6 asm("r6") = ((unsigned)p1 << 22) >> 22;
+#else
+    u32 r7 = p0;
+    u8 *bgOffsets = gBgOffsetsHBlank;
+    s32 r5 = 0;
+    u16 r6 = ((unsigned)p1 << 22) >> 22;
```
Comment (freshollie) [inline src/game/stage/game_7.c:97]: This definitely isn't the matching code, couldn't you just write `r6 = p1`? This to me looks like p1 could be a u8 though

File: src/bg_triangles.c (hunk 2 above)
Comment (JaceCear) [reply src/bg_triangles.c:1]: Yeah, agbcc just doesn't give any warnings on it

File: src/game/multiplayer/mp_player.c (hunk 4 above)
Comment (freshollie) [reply src/game/multiplayer/mp_player.c:753]: !! just means cast to bool

File: src/game/stage/background/zone_3.c (hunk 5 above)
Comment (freshollie) [reply src/game/stage/background/zone_3.c:141]: Ah right

**Review by freshollie — APPROVED** [review]
All good apart from last few questions

**Pull request merged**